Jenkinsfile

```diff
@@ -1031,6 +1032,34 @@ pipeline {
                         }
                     }
                 } // stage('Unit test Bullseye')
+                stage('Unit Test with memcheck') {
+                    when {
+                      beforeAgent true
+                      expression { ! skip_stage('unit-test-memcheck') }
+                    }
+                    agent {
+                        label 'ci_vm1'
+                    }
+                    steps {
+                        unitTest timeout_time: 60,
+                                 ignore_failure: true,
+                                 inst_repos: pr_repos(),
+                                 inst_rpms: unit_packages()
+                    }
+                    post {
+                        always {
+                            // This is only set while dealing with issues
+                            // caused by code coverage instrumentation affecting
+                            // test results, and while code coverage is being
+                            // added.
+                            unitTestPost ignore_failure: true,
+                                         artifacts: ['unit_test_memcheck_logs/*',
+                                                     'unit_memcheck_vm_test/**'],
+                                         valgrind_stash: 'centos7-gcc-unit-memcheck',
+                                         valgrind_pattern: 'unit_test_memcheck_logs/results-memcheck-*.xml'
```

Comment:
Long term, the unitTestPost procedure in pipeline-lib should be able to figure out that these defaults should be set so they are not needed as parameters, especially since in other places these names are being calculated.

However the unitTestPost needs to be backwards compatible with older branches/PRs that expected it to report the valgrind results.   This does not need to be changed for this pass.

Reply:
If you want valgrind to publish the results, the valgrind stash and pattern needed to be added to the valgrind publish step in the last post section of the Jenkinsfile.